### ISSUE TITLE
False positive of `no-callback-in-promise` in `7.2.0`

### ISSUE THREAD

**Issue by zloirock** [eslint-community/eslint-plugin-promise#572]
In many different cases. Example:
```js
'use strict';
var call = require('../internals/function-call');

module.exports = function (record, fn, ITERATOR_INSTEAD_OF_RECORD) {
  var iterator = ITERATOR_INSTEAD_OF_RECORD ? record : record.iterator;
  var next = record.next;
  var step, result;
  while (!(step = call(next, iterator)).done) { // 8:24  error  Avoid calling back inside of a promise  promise/no-callback-in-promise
    result = fn(step.value);
    if (result !== undefined) return result;
  }
};
```

**Comment by scagood** [comment]
Reproduction:

https://eslint-online-playground.netlify.app/#eNp9UsFu2zAM/RVVF9uAI99TBFuxZEAvy5DlVhWBYNOZOpnyJDnLUOTfR9pOsRZoT6b13uN7pPQsY6grOJuud6CeolzKbIggYgq2TtmtxpMJojbOiZUI8HuwAfJMqcpigoDGxaodsE7W44JZWUESjZ1vBuoH596HFEl6JYk8QO1DU4oWS3G/3+zu9tvd4f7bj/3mbn3Yfj3sNl+2u3UhnjUKweaWjEzygbq8zxefKB03Fsu5UFcdBZoaIZzTOMUI898LFBP0JSFxcNPhn5/WgchvckZIxLPlLClf8hSFajzCnFTMap4VR5U6GTcA74NR2/LoI+NmtRIDNtBahKYgXRoC/m9+0Xi5laWE6GjNqvbY2qPqxtuxHa9U9G44WvwefGfptlr6imyiLyZo0U9Ypin1KCFDw/YPbFFVQinF1atOs1d8yFpnUsWr6jqgrE32WGp8pFC9qX+ZI78Vj5RnHF7L9LcHLZdUscKjlkRnoIHTGnpugbWFyJx5X1p+nhJXT+OxlmQJMc1SJkz4h+CbcV9zr8uUl3/xm/vD

**Comment by hsarokina-ita** [comment]
Looks like I'm experiencing the same issue. I noticed that many parts of my code fail linting when they include arguments named `next`, even though they're unrelated to promise usage.

**Comment by hsarokina-ita** [comment]
@dependabot ignore this version

**Comment by kevinoid** [comment]
A reproduction is any `CallExpression` where the callee *is not* named like a callback and an argument *is* named like a callback.  A minimal [repro](https://eslint-online-playground.netlify.app/#eNp9UL1uwyAQfhXEQisleE+XVu0DdA8ZKD67pHAgwFWqyO/eO9tD0yETJ77fu6usxXVwsTEH0OcqD1JNFURtxbumngwaHCZ0zScUn7a+2hA+rPt6KeOD2+ZHcTUohEtYE5mE9Acig1nuJNTgsWmiDH7UccnxMafSRA7T6PG9pOgpd6BXqJW+X6F9XjFlEC6LpIfBTqGJI8d2ndBa83TjtGXVoxqCbV0Bl2IE7KFXp53BE5XKVNCOvHVC6rMsYWT7yWDkgSZWJDSS6Az08P0GmS3QeajMWSSMPa+Nu/PybSRFQm2blAkrfhf8t+4tl6kz33L+BUrplcg=):

```js
function hasCallbackArg(callback) {
  console.log(callback);
}
```

This was regressed by 907753f4b6108ba78b93571a40b6f1384b3c6899

It appears that without the call to `hasPromiseCallback`, there is nothing to constrain the rule to code inside a promise.

CC: @brettz9
